e2e/test/scenarios/native/native-reproductions.cy.spec.ts

```diff
@@ -155,3 +159,43 @@ describe("issue 33327", () => {
     cy.findByTestId("visualization-root").icon("warning").should("not.exist");
   });
 });
+
+describe("issue 33640", () => {
+  const SQL = `
+WHERE [text to be replaced]
+AND y = 20
+`.trim();
+
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should be possible to add a snippet into the native editor without overwriting unrelated text (metabase#33640)", () => {
+    openNativeEditor();
+
+    cy.icon("snippet").click();
+
+    sidebar().last().button("Create a snippet").click();
+
+    modal().within(() => {
+      cy.findByLabelText("Enter some SQL here so you can reuse it later").type(
+        "x = 42",
+      );
+      cy.findByLabelText("Give your snippet a name").type("test");
+      cy.button("Save").click();
+    });
+
+    focusNativeEditor();
+    nativeEditor()
+      .clear()
+      .type(SQL, { timeout: 0 })
+      .type("{selectAll}")
+      .type("{leftarrow}")
+      .type("{rightarrow}".repeat(27), { timeout: 0 })
+      .type("{shift}{leftarrow}".repeat(21), { timeout: 0 })
+      .type("{backspace}");
```

Comment:
I tried simulating selection with:
- `trigger("mousedown")` + `trigger("mousemove")` + `trigger("mouseup")`
- `realMouseDown` + `realMouseMove` + `realMouseUp`
- the abomination below

but I still was unable to reproduce the issue :(

```tsx
    cy.window().then(window => {
      const editor = window.ace.edit(
        window.document.querySelector('[data-testid="native-query-editor"]'),
      );
      editor.selection.setRange(new window.ace.Range(0, 25, 0, 6));
    });
    nativeEditor().type("{backspace}");
```